File: Codeforces/CR 1033/A_Square_of_Rectangles.cpp
/*----------------------------------------
    author:  Imam
    created: 21.06.2025 21:10:12
----------------------------------------*/
#include<bits/stdc++.h>
using namespace std;
inline void solve()
{
    int a,b,c,d,e,f;
    cin>>a>>b>>c>>d>>e>>f;

    if(a==e && a==(b+d+f)){cout<<"YES\n";return;}
    if(a==(c+e) && d==f && a==(b+d)){cout<<"YES\n";return;}
    if(b==f && b==(a+c+e)){cout<<"YES\n";return;}
    if(b==(d+f) && c==e && b==(a+c)){cout<<"YES\n";return;}
    cout<<"NO\n";
}
int main()
{
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr); cout.tie(NULL);
    
    int t=1; cin>>t;
    while(t--) solve();
    return 0;
}

File: Codeforces/CR 1033/A_Square_of_Rectangles.py
def solve():
    l1,b1,l2,b2,l3,b3=map(int,input().split())
    flag=False
    if(l1==(b1+b2+b3) and l1==l2 and l2==l3): flag=True
    if(b1==(l1+l2+l3) and b1==b2 and b2==b3): flag=True
    if(b1==(l1+l2) and b1==(b2+b3) and l2==l3): flag=True
    if(l1==(b1+b3) and l1==(l2+l3) and b2==b3): flag=True
    
    if(flag): print("YES")
    else: print("NO")

t = int(input())
for _ in range(t): solve()